### PR TITLE
 Implement graceful shutdown on Mac 

### LINKED ISSUE #534: Implement graceful shutdown on Mac
I usually unconsciously do Ctrl-C and I found this what I consider a bug or maybe it could just because of my code. I'm not really sure. So when the server is up and running and everything goes perfect right? and when I exit with Ctrl-C it does not finish the exiting process causing port already in use on next start.

```
2024/02/23 10:50:34 ⇨ MongoDB Connected!

   ____    __
  / __/___/ /  ___
 / _// __/ _ \/ _ \
/___/\__/_//_/\___/ v4.11.4
High performance, minimalist Go web framework
https://echo.labstack.com
____________________________________O/_______
                                    O\
⇨ http server started on [::]:1323
^Ccleaning...
see you again~
❯ air

  __    _   ___
 / /\  | | | |_)
/_/--\ |_| |_| \_ v1.50.0, built with Go go1.22.0

watching .
watching cmd
watching cmd/aion
watching internal
watching internal/cmd
watching internal/db
watching internal/handlers
watching internal/middlewares
watching internal/models
watching internal/routes
watching internal/utils
!exclude tmp
building...
running...
2024/02/23 10:50:38 ⇨ MongoDB Connected!

   ____    __
  / __/___/ /  ___
 / _// __/ _ \/ _ \
/___/\__/_//_/\___/ v4.11.4
High performance, minimalist Go web framework
https://echo.labstack.com
____________________________________O/_______
                                    O\
{"time":"2024-02-23T10:50:38.557171+08:00","level":"FATAL","prefix":"echo","file":"api.go","line":"38","message":"listen tcp :1323: bind: address already in use"}
Process Exit with Code: 1
```

My code on graceful shutdown
```
	// Wait for interrupt signal to gracefully shutdown the server with a timeout of 10 seconds.
	// Use a buffered channel to avoid missing signals as recommended for signal.Notify
	quit := make(chan os.Signal, 1)
	signal.Notify(quit, os.Interrupt)
	<-quit
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()

	// Shutdown server
	if err := e.Shutdown(ctx); err != nil {
		e.Logger.Fatal(err)
	}

	// Disconnect from MongoDB
	if err := db.Client.Disconnect(context.Background()); err != nil {
		log.Fatal("⇨ Error disconnecting to MongoDB: ", err)
		log.Fatal(err)
	}
```

### PR DESCRIPTION
**Pull request by xiantang**
close https://github.com/air-verse/air/issues/534

### CODE REVIEW

File: runner/engine.go [hunk 1]
```diff
@@ -536,6 +537,7 @@ func (e *Engine) runBin() error {
 			case <-killCh:
 				return
 			default:
+				e.procKillWg.Add(1)
```
Comment (nakkamarra) [inline runner/engine.go:540]: What I don't understand is why the WaitGroup passed to `killFunc` on line 553 doesn't seem to be working properly, I'd expect that to have waited for the process to be killed.

Comment (xiantang) [reply runner/engine.go:540]: because killFunc is running in a goroutine

Comment (nakkamarra) [reply runner/engine.go:540]: No that's expected of course, but the WaitGroup is expects as an argument is defined on 522 and is only ever `Wait`ed on line 528, so perhaps waiting on that wasn't working as expected or the `Wait` wasn't being called every time. To me, at least, figuring that out seems like a better solution than adding yet another WaitGroup

Comment (xiantang) [reply runner/engine.go:540]: when i remove the wg.Done(), the we still can exit. i think it's pointer issue. because we put the pointer into funtion

**Review by nakkamarra — APPROVED** [review]

**Pull request merged**